Finished Part 2

dantevibes opened the pull request:
Changed some function names and variables to make more sense. Did less of that near the end due to time constraints. Most remaining compiler warnings were from variables that I didn't use in real functions, but I used in pseudo-code to display understanding. I didn't want to leave many functions blank.

dantevibes committed:
Finished Part 2. Changed some function names and variables to make more sense. Did less of that near the end due to time constraints. Most remaining compiler warnings were from variables that I didn't use in real functions, but I used in pseudo-code to display understanding. I didn't want to leave many functions blank.

matkatmusic commented on main.cpp:
```diff
@@ -125,419 +152,409 @@ struct CarWash
+
+bool Octopus::catchAFish(int whichArm, int distanceToFish)
+{
+    bool fishWasCaught = false;
+    if (whichArm + distanceToFish < 10)
+    {
+        fishWasCaught = true;
+    }
+
+    return fishWasCaught;
```
you can just write 
`return (whichArm + distanceToFish < 10);`
the stuff between the `( )` will be evaluated and the result of that evaluation will be the value that is returned.
it's nice and concise.
- [x] revise

matkatmusic reviewed (changes requested):
You have a bunch of warnings that you need to resolve. 
See my Slack message.

dantevibes committed:
cleared up all warnings. Function names do not perfectly relate to the action being executed.

matkatmusic commented on main.cpp:
```diff
@@ -125,419 +152,429 @@ struct CarWash
+bool Airplane::angleWingsForTakeoff(int wingAngleDegree)
+{
+    wingAngleDegree += 3;
+    bool isFlying = false;
+    if( engineHorsePower > 2000)
+    {
+        isFlying = true;
+    }
+    return isFlying;
```
you can write `return engineHorsePower > 2000;` instead of writing this `if()` block
- [x] revise

matkatmusic commented on main.cpp:
```diff
@@ -125,419 +152,429 @@ struct CarWash
-    3)  Sweep up/down
- */
+
+void Oscillator::outputSignal(char channel, int amplitude)
+{
+    for(int i = 0; i < 2; i++)
+    {
+        channel = 'L';
+        amplitude = 75 * i;
+    }
+}
```
FYI: you're overwriting the `channel` function parameter.

matkatmusic commented on main.cpp:
```diff
@@ -125,419 +152,429 @@ struct CarWash
+    for(float f = 0.0f; f < sweepTimeInMillis; f += 0.5f)
+    {
+        cutoffFreq = startFreq;
+        if(cutoffFreq < endFreq) cutoffFreq++;
+
+        else  cutoffFreq--;
```
always prefer Pre-increment instead of Post-increment. 
if you need an explanation, send me a message in slack.
Also, it's generally more readable if the `if()` body is on its own line when its a single-line expression:
```
if( expr )
    singleLineExpression
else
    singleLineExpression
```
it's just easier to *see* the logic.
- [x] revise

matkatmusic commented on main.cpp:
```diff
@@ -125,419 +152,429 @@ struct CarWash
+{
+    outputLevel = outputLevel/denominator;
+}
+void Amplifier::addDrive(float driveAmount, Filter inputFilter, Filter outputFilter)
+{
+    outputFilter.cutoffFreq = inputFilter.cutoffFreq;
+    driveLevel += driveAmount;
+}
```
When you learn about References, this will be perfectly fine code.  However, these are not references (and you can't use references yet because they haven't been covered).  these function parameters stop existing when the function ends, which means your changes to `outputFilter.cutoffFreq` aren't preserved.
- [x] revise.  try making `outputFilter` a member variable of your Amplifier so the changes will be preserved.

matkatmusic commented on main.cpp:
```diff
@@ -125,419 +152,429 @@ struct CarWash
+}
+void Delay::moveFeedbackPrePost(bool preWet)
+{
+    if(preWet)
+    {
+        feedbackLevel ++;
+    }
+    else
+    {
+        feedbackLevel --;
+    }
```
prefer pre-increment over post-increment
- [x] revise

matkatmusic commented on main.cpp:
```diff
@@ -107,8 +107,35 @@ struct CarWash
     you should be able to deduce the return type of those functions based on their usage in Person::run()
     You'll need to insert the Person struct from the video in the space below.
  */
+struct Person 
+{
+    int age;
+    int height;
+    float hairLength;
+    float GPA;
+    unsigned int SATScore;
+
+    void run(int howFast, bool startWithLeftFoot);
+};
```
You forgot to copy the `run()` implementation from the video.
- [x] revise

dantevibes committed:
-Better single-line formatting
-pre-increment instead of post
-storing values of function parameters
-added run function to Person
-changed functions and format of Foot type to better suit run function.
